Slider/Assets/Scripts/Audio/Dialogue/SentenceVocalizer.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Analytics;
+
+[System.Serializable]
+public class SentenceVocalizer : IVocalizerComposite<WordVocalizer>
+{
+    private static readonly string endingsStr = @"(?<=[,.?!])";
+    private static readonly char[] endings = { ',','.','?','!' };
+    private static readonly HashSet<char> endingsSet = new(endings);
+    /// <summary>
+    /// Questions no longer tend to tone upwards when the 6W words are in the same clause
+    /// </summary>
+    private static readonly string[] questionNegation = { "who", "what", "when", "where", "why", "how" };
+
+    public List<WordVocalizer> words;
+    public char punctuation;
+    public List<WordVocalizer> Vocalizers => words;
+    public bool IsEmpty => words.Count == 0;
+    WordVocalizer IVocalizerComposite<WordVocalizer>.Current { get => _Current; set => _Current = value; }
+    private WordVocalizer _Current;
+    VocalizerCompositeStatus IVocalizerComposite<WordVocalizer>.Status { get => _Status; set => _Status = value; }
+    private VocalizerCompositeStatus _Status;
+
+    public enum Intonation { flat, up, down };
+    public Intonation intonation;
+
+    public static List<SentenceVocalizer> Parse(string paragraph)
+    {
+        string[] clauses = Regex.Split(CleanUp(paragraph), endingsStr); // split while keeping the delimiter
+        var vocalizers = new List<SentenceVocalizer>(clauses.Length);
+        foreach (var clause in clauses)
+        {
+            if (clause.Length <= 1) continue;
+            else
+            {
+                SentenceVocalizer sv = new(clause);
+                if (!sv.IsEmpty) vocalizers.Add(sv);
+            }
+        }
+        return vocalizers;
+    }
+
+    /// <param name="clause">Alphanumeric words separated by space with the last character being one of the characters in "endings"</param>
+    private SentenceVocalizer(string clause)
+    {
+        words = new List<WordVocalizer>();
+        int endingTrim = endingsSet.Contains(clause[^1]) ? 1 : 0; // remove ending character if it is a punctuation
+        punctuation = endingTrim > 0 ? clause[^1] : '.'; // default to period when no punctuation
+
+        foreach(string word in clause.Substring(0, clause.Length - endingTrim).Split(' ', System.StringSplitOptions.RemoveEmptyEntries))
+        {
+            WordVocalizer wv = new(word);
+            if (!wv.IsEmpty) words.Add(wv);
+        }
+
+        if (punctuation == '?')
+        {
+            intonation = Intonation.up;
+            foreach (var keyword in questionNegation)
+            {
+                if (words[0].characters.StartsWith(keyword))
+                {
+                    intonation = Intonation.down;
```

Comment:
do you think it would be worth it to add an option to manually override the ending intonation in npc dialogue settings (like where we set animations/emotes)